server.py

```diff
@@ -155,6 +155,10 @@ async def get_root(request):
         def get_embeddings(self):
             embeddings = folder_paths.get_filename_list("embeddings")
             return web.json_response(list(map(lambda a: os.path.splitext(a)[0], embeddings)))
+        
+        @routes.get("/models")
+        def list_model_types(request):
+            return web.json_response(list(folder_paths.folder_names_and_paths))
```

Comment:
probably only need to grab the keys here? the values are local system paths which shouldn't be sent over network api

Reply:
The endpoint should be something like `GET /models/paths` or `GET /models/folders`. 
Since it is only returning the types of models or available folders.

Reply:
> probably only need to grab the keys here? the values are local system paths which shouldn't be sent over network api

Using `list()` does only grab the keys, I might need to clarify that in code.

![image](https://github.com/user-attachments/assets/a86528c4-9e84-4c69-9d3b-3c640ee24c59)

Reply:
> The endpoint should be something like `GET /models/paths` or `GET /models/folders`. Since it is only returning the types of models or available folders.

Wouldn't that interfere with if someone were to have an extra model path named either one of those? Probably unlikely but I think it would be better if there wasn't a chance for route conflict.

I can change the route if needed its just that setting it to `/models` seemed the most logical.

Reply:
yes that format would be a conflict. It works as `/models`, if a different name is needed it'd have to be eg `/model_folders` or something